k8s/distributed-chroma/templates/compaction-service-memberlist-cr.yaml

```diff
@@ -0,0 +1,79 @@
+# These kubernetes manifests are UNDER ACTIVE  DEVELOPMENT and are not yet ready for production use.
+# They will be used for the upcoming distributed version of chroma. They are not even ready
+# for testing yet. Please do not use them unless you are working on the distributed version of chroma.
+
+apiVersion: chroma.cluster/v1
+kind: MemberList
+metadata:
+  name: compaction-service-memberlist
+  namespace: {{ .Values.namespace}}
+spec:
+  members:
+
+---
+
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: compaction-service-memberlist-readerwriter
+rules:
+- apiGroups:
+  - chroma.cluster
+  resources:
+  - memberlists
+  verbs:
+  - get
+  - list
+  - watch
+  # TODO: FIX THIS LEAKY PERMISSION
+  - create
+  - update
+  - patch
+  - delete
+
+---
+
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: sysdb-compaction-service-memberlist-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: compaction-service-memberlist-readerwriter
+subjects:
+- kind: ServiceAccount
+  name: sysdb-serviceaccount
+  namespace: {{ .Values.namespace }}
+
+---
+
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  # Awkward name, but this lets the query-service-serviceaccount read
+  # the query-service-memberlist.
+  name: compaction-service-compaction-service-memberlist-binding
+roleRef:
```

Comment:
should we share permissions across the query/compaction roles ?

Reply:
are you asking if both service accounts should be bound to the exact same cluster role? Probably not in the long run but I'm fine with it for now -- very low risk it causes problems since the query service literally doesn't have code to modify these